.github/workflows/tests.yml

```diff
@@ -2,7 +2,28 @@ name: Tests
 on: pull_request
 
 jobs:
+  linters:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+
+      - name: Setup Python
+        uses: actions/setup-python@v4
+        with:
+          python-version: '3.10'
+
+      - name: Install dependencies
+        run: |
+          pip install poetry
+          poetry config virtualenvs.create false
+          poetry install --no-root
+
+      - name: Run pre-commit
+        run: pre-commit run --all-files
```

Comment:
I'm not sure this works. The pre-commit fixes the files locally in the GitHub action but the CI won't fail if there are error, won't it? I guess we need to run `black`, `isort` (and eventually other programs) with a `--check` option (or some equivalent).

I'm not sure I know how to solve this, but maybe, we'll need to go back to running `black --check` and `isort --check` manually?